engine/runtime/src/test/java/org/enso/compiler/EnsoCompilerTest.java

```diff
@@ -557,8 +608,75 @@ public void testBalanceLowerCase() throws Exception {
       """);
   }
 
+  @Test
+  public void testVectorVectorSimple() throws Exception {
+    parseTest("""
+    type Vector
+        build : Matrix Any Decimal
+    """);
+  }
+
+  @Test
+  public void testVectorVectorAny() throws Exception {
+    parseTest("""
+    type Vector
+        build : Standard.Base.Vector.Matrix Standard.Base.Any Standard.Base.Decimal
+    """);
+  }
+
+  @Test
+  public void testCaseOfVector() throws Exception {
+    parseTest("""
+        m other = case other of
+            _:Vector.Vector -> 0
+        """);
+  }
+
+  @Test
+  public void testOperatorSectionRight() throws Exception {
+    parseTest("""
+    type Filter_Condition
+        to_predicate self = case self of
+            Less value -> <value
+    """);
+  }
+
+  @Test
+  public void testAutoScope() throws Exception {
+    parseTest("""
+    fn that_meta =
+        c_2 = that_meta.constructor ...
+        """);
+  }
+
+  @Test
+  public void testTextArrayType() throws Exception {
+    parseTest("""
+    type Connection
+        table_types : [Text]
+    """);
+  }
+
+  @Test
+  public void testListBody() throws Exception {
+    parseTest("""
+          list directory name_filter=Nothing recursive=False =
+              new directory . list name_filter=name_filter recursive=recursive
+                  """);
+  }
+
+  @Test
+  @Ignore
+  public void testLambdaBody() throws Exception {
+    parseTest("""
+    list =
+        all_files.filter file->
```

Comment:
@kazcw & co. I am trying to execute `main = IO.println "Hello"` and as far as I can say this is the main blocker. I am not aware of any reasonable workaround - right now I just replace the body with a dummy identifier `no_body_for_lambda`, but I am not sure if that's going to be enough to run the _"Hello world"_ example by Monday.

Reply:
Thank you for the test case. I'll try to get this working today.
We don't need workarounds--if it parses wrong and produces the right IR, `parser` and `TreeToIr` are both wrong.

Reply:
Fix is in #3795